rero_ils/modules/items/api/api.py

```diff
@@ -135,6 +139,26 @@ def reasons_not_to_delete(self):
             )
         return cannot_delete
 
+    def in_collection(self, **kwargs):
+        """Get published collection pids for current item."""
+        from ...collections.api import CollectionsSearch
+        output = []
+        search = CollectionsSearch() \
+            .filter('term', items__pid=self.get('pid')) \
+            .filter('term', published=True) \
+            .sort({'title_sort': {"order": "asc"}}) \
+            .params(preserve_order=True) \
+            .source(['pid', 'organisation', 'title'])
+        for record in search.scan():
+            organisation = Organisation.get_record_by_pid(
+                record.organisation.pid)
```

Comment:
for each hit, you load a organisation ; For performance it's not very good because you will often load always the same organisation. Create an `orgs = {}` outside of the loop:
```python
for record in search.scan():
    if record.organisation.pid not in orgs:
       orgs[record.organisation.pid] = Organisation.get_record_by_pid(record.organisation.pid)
    output.append({
        'pid': record.pid,
        'title': record.title,
        'viewcode': orgs[record.organisation.pid].get('code')
     })
```